scripts/build-icons.ts

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright 2024 Mia srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* eslint-disable no-console */
+
+import { Cheerio, Element, load } from 'cheerio'
+import { type TarOptionsWithAliasesAsyncNoFile, extract } from 'tar'
+import { IconTree } from 'react-icons'
+import camelcase from 'camelcase'
+import fs from 'fs/promises'
+import { glob } from 'glob'
+import https from 'https'
+import path from 'path'
+import url from 'url'
+
+const REACT_ICONS_VERSION = '5.2.1'
+const REACT_ICONS_PACKAGES = ['lib', 'ai', 'fi', 'pi']
+
+const dirname = path.dirname(url.fileURLToPath(import.meta.url))
+const rootDir = path.resolve(dirname, '..')
+const outDir = path.resolve(rootDir, 'icons')
+
+async function setup(): Promise<void> {
+  const outDirContent = await fs.readdir(outDir)
+
+  const promises = outDirContent.reduce<Promise<void>[]>((acc, cur) => {
+    if (cur !== 'package.json') {
+      const pathToRemove = path.resolve(outDir, cur)
+      return [...acc, fs.rm(pathToRemove, { recursive: true, force: true })]
+    }
+
+    return acc
+  }, [])
+
+  await Promise.all(promises)
+}
+
+async function downloadReactIconsPackage(): Promise<void> {
+  const pkgUrl = `https://github.com/react-icons/react-icons/releases/download/v${REACT_ICONS_VERSION}/react-icons-all-files-${REACT_ICONS_VERSION}.tgz`
```

Comment:
We need to download the tar directly from GitHub since the "all files" version of react-icons cannot be loaded on NPM given the high number of files (https://github.com/react-icons/react-icons/issues/593)